EIPS/eip-4519.md

```diff
@@ -0,0 +1,239 @@
+---
+eip: 4519
+title: Smart Non-Fungible Token (SmartNFT)
+description: A standard interface of smart non-fungible tokens representing smart assets that can generate their own blockchain accounts and obey users.
+author: Javier Arcenegui (@Hardblock-IMSE-CNM), Rosario Arjona <arjona@imse-cnm.csic.es>, Roberto Román <roman@imse-cnm.csic.es>, Iluminada Baturone <lumi@imse-cnm.csic.es>
+discussions-to: https://ethereum-magicians.org/t/new-proposal-of-smart-non-fungible-token/7677
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-12-03
+requires: 165, 721
+---
+
+## Abstract
+The novelty introduced in this EIP is the proposal of a Smart Non-Fungible Token (SmartNFT) to represent a physical smart asset, such as an IoT device. A SmartNFT is physically bound to a smart asset. A Smart Asset can hold an address to participate in the Ethereum chain. It also serves as a utility to the user as it can establish secure communication with their owners and other Ethereum users. Likewise, it can operate dynamically with several operating modes associated with its token states. For example, an IoT device capable of generating its address from its private key and is also the sole owner of this private key. The IoT device does not store the private key but uses a Physical Unclonable Function (PUF) that allows recovering the private key. SmartNFTs extend [ERC-721](./eip-721.md) non-fungible tokens (which only allow representing assets by a unique identifier, as a possession of an owner).
+ 
+## Motivation
+There are nowadays many smart assets, like IoT devices, that can interact actively with other blockchain participants, that is, they can have an Ethereum account. Hence, we propose the SmartNFTs, which have the attribute "asset" to represent the address associated with the physical smart asset. Besides, a smart asset is not only a possession of an owner but also an active agent that obeys to a user to carry out certain tasks in an application. Hence, the SmartNFTs have the NFT attribute "user", which is the address of the user of the smart asset. In order to represent the secure dynamic activity of the smart asset, six more attributes are proposed: "timestamp", "timeout", "state", "hashK_OA", "hashK_UA", and "dataEngage".  The attribute "timestamp" registers in the blockchain whenever the smart asset checks it is bound to its token. The attribute "timeout" is the maximum delay time established for the asset to prove again the bond. Token "state" is included to trace the operating mode of the smart asset while "hashK_OA", "hashK_UA", and "dataEngage" are public data added to trace if secure communication channels are established between the smart asset and its owner and user. SmartNFTS extend ERC-721 NFTs, which are designed to represent passive assets, which only have an owner that owns the token, can transfer ownership, and can approve others to act in its name. 
```

Comment:
A lot of what's in here seems to be describing why SmartNFTs have certain attributes. The "Motivation" section should be describing why the EIP as a whole exists, and the "Rationale" section describes the reasoning behind particular choices made _in_ the EIP.